x-pack/plugin/apm-data/src/internalClusterTest/java/org/elasticsearch/xpack/apmdata/APMRolloverIT.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.apmdata;
+
+import org.elasticsearch.action.ActionFuture;
+import org.elasticsearch.action.datastreams.CreateDataStreamAction;
+import org.elasticsearch.action.datastreams.GetDataStreamAction;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.cluster.ClusterChangedEvent;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.ComposableIndexTemplate;
+import org.elasticsearch.cluster.metadata.DataStream;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.datastreams.DataStreamsPlugin;
+import org.elasticsearch.index.mapper.extras.MapperExtrasPlugin;
+import org.elasticsearch.ingest.common.IngestCommonPlugin;
+import org.elasticsearch.ingest.geoip.GeoIpDownloaderTaskExecutor;
+import org.elasticsearch.ingest.geoip.IngestGeoIpPlugin;
+import org.elasticsearch.ingest.useragent.IngestUserAgentPlugin;
+import org.elasticsearch.painless.PainlessPlugin;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.aggregatemetric.AggregateMetricMapperPlugin;
+import org.elasticsearch.xpack.analytics.AnalyticsPlugin;
+import org.elasticsearch.xpack.constantkeyword.ConstantKeywordMapperPlugin;
+import org.elasticsearch.xpack.core.XPackPlugin;
+import org.elasticsearch.xpack.core.XPackSettings;
+import org.elasticsearch.xpack.ilm.IndexLifecycle;
+import org.elasticsearch.xpack.ilm.history.ILMHistoryTemplateRegistry;
+import org.elasticsearch.xpack.stack.StackPlugin;
+import org.elasticsearch.xpack.stack.StackTemplateRegistry;
+import org.elasticsearch.xpack.wildcard.Wildcard;
+import org.junit.After;
+import org.junit.Before;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.hamcrest.Matchers.endsWith;
+import static org.hamcrest.Matchers.hasSize;
+
+@SuppressWarnings("resource")
+// the Painless compiler requires the "createClassLoader" permission
+@ESTestCase.WithoutSecurityManager
```

Comment:
This strikes me as something that needs to be cleaned up before the PR can be accepted - There are only a small handful of tests that need this currently and they seem to be much lower level than what this is testing. Is there a way we could solve this problem some other way? I'm not entirely convinced that this test case makes sense where it is, but if we keep it around, I think this would need a more proper fix.

Reply:
I agree. I looked for something, assuming there is a test setup I need to make to prevent this, but couldn't find one so far. I'll keep looking.